### ISSUE TITLE
SDL.pumpEvents seems to be required in mainLoop for Windows 10

### ISSUE THREAD

**Issue by JD95** [fjvallarino/monomer#66]
I can't say exactly what changed on my system to cause this, but my monomer projects suddenly started hanging immediately after launch. Even reverting to older commits which I know worked or cloning monomer and building the example projects had the same behavior.

I did a bit of digging and realized that SDL.pollEvents wasn't returning anything after the first frame.

The fix ended up being a call to SDL.pumpEvents before starting the new frame in Monomer.Main.Core.mainLoop

```haskell
unless shouldQuit (SDL.pumpEvents *> mainLoop window fontManager config newLoopArgs)
```

I'm hesitant to make a PR given that I'm not sure what exactly changed to cause this, but it is consistently reproducible. 

One thought I had was the SDL2 version which is `mingw-w64-x86_64-SDL2 2.0.20-1`, but I'm at a bit of a loss otherwise.

**Comment by fjvallarino** [comment]
Hi! This is interesting. I'm also leaning on this being an issue with SDL2, or maybe the bindings.

I noticed a new version of the Haskell [package](https://hackage.haskell.org/package/sdl2-2.5.3.1) was released about a week ago. If you are not using the old Stack LTS (i.e., using Cabal), maybe it's worth trying to fix the `sdl2` version to 2.5.3.0, which was released more than a year ago.

I took a quick look, and switching to a previous version of the SDL2 binary with `Pacman` does not seem to be easy (in case this is the source of the problem).

My Windows box is currently in a bad state; I'll take a look as soon as I get it back to working properly.

**Comment by fjvallarino** [comment]
It seems you can manually download a specific version of a library from [here](http://repo.msys2.org/mingw/mingw64) and install it with:

```shell
stack exec -- pacman -U mingw-w64-x86_64-SDL2-2.0.14-2-any.pkg.tar.zst
```

Removing the current version can be done with:

```shell
stack exec -- pacman -R mingw-w64-x86_64-SDL2
```

**Comment by fjvallarino** [comment]
I upgraded to SDL2 2.0.20, released eight days ago, and I'm having the same issue on macOS. I think it's related to [this PR](https://github.com/libsdl-org/SDL/pull/4794), where it's also discussed that calling SDL_PollEvent in a loop is not a good idea (which is what `pollEvents` does internally). 

I'll take a look to see the recommended way of polling events and implement it.

Thanks for reporting!

**Comment by fjvallarino** [comment]
I went ahead and followed your proposed solution. I looked into implementing the idea suggested [here](https://github.com/libsdl-org/SDL/pull/4794#issuecomment-941569077), but I'm not sure it's worth it. If the need arises in the future, it can be reviewed.

This is already merged to main.

Thanks for the diagnostic and the solution!

**Comment by fjvallarino** [comment]
I'll close this issue, since the fix has already been merged to main. It will be included in the 1.3 release (to be released soon). Thanks!
